apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -81,6 +81,13 @@ type OpenTelemetryCollectorSpec struct {
 	// +optional
 	// +listType=atomic
 	Ports []v1.ServicePort `json:"ports,omitempty"`
+	// ContainerPorts allows a set of ports to be exposed by the underlying v1.Container. By default, the operator
+	// will attempt to infer the required ports by parsing the .Spec.Config property but this property can be
+	// used to open additional ports that can't be inferred by the operator, like for custom receivers.  Any ports
```

Comment:
> like for custom receivers.

I would assume that custom receiver ports should be defined in the `spec.ports` and the container ports are rather used for the ports that should not be exposed in the service. 


Do we even need to have service ports and container ports? We could use the service ports to open the ports on the container. Is there any use-case when container ports != service ports?

Reply:
I'm not sure - I included this for the flexibility and because the types are different.  I can change it so we parse the configured service ports into container ports if that's the direction we want to go in?

Reply:
@pavolloffay, do you want me to change the logic to parse the container ports from the list of service ports? Or are you waiting for others' input on this?

Reply:
If there is no hard requirement to expose additional configuration in the CR then I would opt-in for deriving the ports automatically for now